pkg/minikube/driver/driver.go

```diff
@@ -164,14 +164,23 @@ func FlagDefaults(name string) FlagHints {
 }
 
 // Choices returns a list of drivers which are possible on this system
-func Choices() []registry.DriverState {
+func Choices(vm bool) []registry.DriverState {
+	var drivers []registry.DriverState
 	options := registry.Available()
-
+	if vm {
+		for _, ds := range options {
+			if IsVM(ds.Name) {
```

Comment:
Suggestion: Now that I see how this looks, rather than applying the filter in the Choices function, could you pass the value to `registry.Available` and add the filtering in that function instead?

https://github.com/kubernetes/minikube/blob/07e8420c2ab57de73da61407e5dff777552b957b/pkg/minikube/registry/global.go#L62

This would be a slight performance improvement, as `registry.Available()` would not have to call`driver.Status()` on non-VM drivers if the result is going to be ignored anyways.

Reply:
sure I will make the suggested changes soon

Reply:
> Need any help with this PR?

@tstromberg I have made the changes. I had to write **IsVM** function separately in registry folder to avoid "import cycle error" b/w driver and registry.